### PR TITLE
DB-8714 Factor out common SpliceDefaultCompactor functions (2.8)

### PR DESCRIPTION
**Pull request by arnaud-splice**
With this commit, we only factor out the functions of
SpliceDefaultCompactor which are absolutely identical accross all
hbase_sql/hbase<...> versions.

A later commit may be necessary to factor out the meaty part as well.

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

File: hbase_sql/hbase1.1.1/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java [hunk 1]
```diff
@@ -16,83 +16,45 @@
...
-
-        if (LOG.isDebugEnabled()) {
-            SpliceLogUtils.debug(LOG, "Initializing compactor: region=%s", ((HStore)this.store).getHRegion());
-        }
     }
 
     public SpliceDefaultCompactor(final Configuration conf, final Store store, long smallestReadPoint) {
-        this(conf, store);
-        this.smallestReadPoint = smallestReadPoint;
+        super(conf, store);
```
Comment (carolp-503) [inline hbase_sql/hbase1.1.1/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java]: Resolve @yxia92 's comment on Master here before merging.

Comment (arnaud-splice) [reply hbase_sql/hbase1.1.1/src/main/java/com/splicemachine/compactions/SpliceDefaultCompactor.java]: Done.

**Comment by arnaud-splice** [comment]
> * All comments are resolved.

Done
> * @dgomezferro has reviewed master, and any corresponding changes are made both in Master and here.

I will wait for his review 
> * Pending changes for 3.0 branch can be merged separately, but we should make the same changes before moving onto the next phase of workload isolation/swimlane for compaction

No changes seem to be required in 3.0 to mirror that PR as the folder structure is different there and no version specific code exists.

**Pull request merged**